### PR TITLE
Add snapshot metadata class and a serializer.

### PR DESCRIPTION
**Pull request by mansu**
Add a snapshot metadata class and a serializer/deserializer which can store metadata in ZK.

This is the first diff in a series of diffs.

### CODE REVIEW

**Review by vthacker — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -153,24 +153,25 @@
             <version>${jackson.version}</version>
         </dependency>
 
+        <!-- YAML config parsing dependencies -->
```
Comment (bryanlb) [inline pom.xml:156]: Fwiw these comments tend to go stale pretty quickly as the project grows. For example - Jackson `core` and `databind` are now general serialization dependencies used for both json and yaml parsing. What will happen over time is folks will use a dependency we've already pulled in (ie, `commons-text`) and not realize there's an associated POM comment that is no longer valid.

**Comment by bryanlb** [comment]
Can we get some of these commits squashed? There seems to be a lot of variations of "add test" without any additional detail.

**Comment by vthacker** [comment]
> Can we get some of these commits squashed? There seems to be a lot of variations of "add test" without any additional detail.

image

I think we can do that while committing - It's the default option for me atleast

File: pom.xml (hunk 1 above)
Comment (mansu) [reply pom.xml:156]: I agree this is true with code comments but for the dependencies it seemed to work well atleast so far. I am flexible either ways but these 2 dependencies are related to yaml, so I thought I call them out explicitly.

**Comment by mansu** [comment]
Sure, I can squash and merge the commits.

**Review by bryanlb — APPROVED** [review]

**Pull request merged**